components/scream/scripts/gen_boiler.py

```diff
@@ -0,0 +1,1878 @@
+from utils import expect, get_git_toplevel_dir
+
+from collections import OrderedDict
+import pathlib, re, os
+
+#
+# Global hardcoded data
+#
+
+# Templates: maps piece name to generic file text
+FILE_TEMPLATES = {
+    "cxx_bfb_unit_impl": lambda phys, sub, gen_code:
+"""#include "catch2/catch.hpp"
+
+#include "share/scream_types.hpp"
+#include "ekat/ekat_pack.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "physics/{physics}/{physics}_functions.hpp"
+#include "physics/{physics}/{physics}_functions_f90.hpp"
+
+#include "{physics}_unit_tests_common.hpp"
+
+namespace scream {{
+namespace {physics} {{
+namespace unit_test {{
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::{test_data_struct} {{
+
+{gen_code}
+
+}};
+
+}} // namespace unit_test
+}} // namespace {physics}
+}} // namespace scream
+
+namespace {{
+
+TEST_CASE({sub}_bfb, "[{physics}_functions]")
+{{
+  using TestStruct = scream::{physics}::unit_test::UnitWrap::UnitTest<scream::DefaultDevice>::{test_data_struct};
+
+  TestStruct::run_bfb();
+}}
+
+}} // empty namespace
+""".format(physics=phys, sub=sub, test_data_struct=get_data_test_struct_name(sub), gen_code=gen_code),
+
+    "cxx_func_impl": lambda phys, sub, gen_code:
+"""#ifndef {phys_upper}_{sub_upper}_IMPL_HPP
+#define {phys_upper}_{sub_upper}_IMPL_HPP
+
+#include "{physics}_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {{
+namespace {physics} {{
+
+/*
+ * Implementation of {physics} {sub}. Clients should NOT
+ * #include this file, but include {physics}_functions.hpp instead.
+ */
+
+template<typename S, typename D>
+{gen_code}
+
+}} // namespace {physics}
+}} // namespace scream
+
+#endif
+""".format(physics=phys, sub=sub, gen_code=gen_code, phys_upper=phys.upper(), sub_upper=sub.upper()),
+
+    "cxx_eti": lambda phys, sub, gen_code: gen_code
+}
+
+# piece map. maps the name of a piece of boilerplate that needs to be genereate to:
+#   (filepath (relative to cxx_root))
+FILEPATH, FILECREATE, INSERT_REGEX, ID_SELF_BEGIN_REGEX, ID_SELF_END_REGEX, DESC = range(6)
+PIECES = OrderedDict([
+    ("f90_c2f_bind", (
+        lambda phys, sub, gb: "{}_iso_c.f90".format(phys),
```

Comment:
Since we are (re-)establishing a pattern, could we call the bridge files `{}_f2c.f90` rather than `{}_iso_c.f90` (and similarly for the other way around)? I am never sure (and I might have gotten it wrong here too) which direction are `{}_iso_[c|f].f90` going. Imho, `f2c` and `c2f` are way clearer in explaining who calls what.

Reply:
Yes, that should be easy if others agree.

Reply:
I agree that _f2c.f90 would be better than iso_c.f90!